include/gz/rendering/base/BaseCamera.hh

```diff
@@ -498,6 +501,32 @@ namespace gz
       return RenderWindowPtr();
     }
 
+    //////////////////////////////////////////////////
+    template <class T>
+    math::Matrix3d BaseCamera<T>::CameraIntrinsicMatrix() const
+    {
+      const math::Matrix4d& projectionMat = this->ProjectionMatrix();
+
+      double right = this->ImageWidth();
+      double left = 0.0;
+      double top = this->ImageHeight();
+      double bottom = 0.0;
+
+      double inverseWidth = 1.0 / (right - left);
+      double inverseHeight = 1.0 / (top - bottom);
+
+      double fX = projectionMat(0, 0)/ (2*inverseWidth);
+      double fY = projectionMat(1, 1) / (2*inverseHeight);
+      double cX = -((projectionMat(0, 2) -
+          ((right + left) / (right - left))) * (right - left)) / 2;
+      double cY = this->ImageHeight() + ((projectionMat(1, 2) -
+          ((top + bottom) / (top - bottom))) * (top - bottom)) / 2;
```

Comment:
Do you have an online reference for the math? It looks like we can simplify, considering we have the image width and height directly, e.g.

```suggestion
      const double width = static_cast<double>(this->ImageWidth());
      const double height = static_cast<double>(this->ImageHeight());

      double fX = (projectionMat(0, 0) * width) / 2.0;
      double fY = (projectionMat(1, 1) * height) / 2.0;
      double cX = (-1.0 * width * (projectionMat(0, 2) - 1.0) / 2.0;
      double cY = height + (height * (projectionMat(1, 2) - 1)) / 2.0;
```

Reply:
I agree we can simplify but the only I had it this way so that it is easier to relate to the original `ProjectionMatrix()` calculation since that's how ogre and gazebo is using the term names. Here are some links for reference, though there is no specific link I think to calculate the intrinsics from `ProjectionMatrix()`, I just decoupled it. 

I can simplify the calculation as well not sure. 

https://ogrecave.github.io/ogre/api/13/class_ogre_1_1_math.html#a23c7798ef5a11dea62318b5be2983c2d

Reply:
It would be nice to simplify the math here, and add the link as reference in a comment.

Reply:
Done https://github.com/gazebosim/gz-rendering/pull/700/commits/083883f1a90ca02c2892cfb52824f9b5121cf49d